File: Assets/common/script/ship/SubsystemCollider.js
#pragma strict

var targetSystem : String;
private var target : BaseSubsystem;
private var theShip : GameObject;

function Start () {
	theShip = GameObject.Find("TheShip");

	target = theShip.GetComponent(targetSystem);
	
	
}

function Update () {

}

function OnCollisionEnter(collision : Collision){
	if(collision.transform != theShip.transform){
		target.applyImpactDamage(collision.impactForceSum.magnitude);
		Debug.Log(collision.impactForceSum.magnitude);
		Debug.Log(collision.other.name);
	}
}

File: Assets/common/script/ship/SubsystemCollider.cs
using UnityEngine;
using System;


public class SubsystemCollider:MonoBehaviour{
	
	public string targetSystem;
	BaseSubsystem target;
	GameObject theShip;
	
	public void Start() {
		theShip = GameObject.Find("TheShip");
	
		//target = theShip.GetComponent<TargettingSystem>();
		
		
	}
	
	public void Update() {
	
	}
	
	public void OnCollisionEnter(Collision collision){
		if(collision.transform != theShip.transform){
			target.applyImpactDamage(collision.impactForceSum.magnitude);
			UnityEngine.Debug.Log(collision.impactForceSum.magnitude);
			UnityEngine.Debug.Log(collision.other.name);
		}
	}
}
